src/odemis/gui/comp/canvas.py

```diff
@@ -1548,8 +1548,17 @@ def on_dbl_click(self, evt):
 
             # recompute the view
             offset = (-shift[0] / self.scale, shift[1] / self.scale)
-            new_pos = (self.p_buffer_center[0] + offset[0],
-                       self.p_buffer_center[1] + offset[1])
+            new_pos = [self.p_buffer_center[0] + offset[0],
+                       self.p_buffer_center[1] + offset[1]]
+
+            # Update the position within the allowed limits. The Streamview class logs if the movement is limited.
+            stage_limits = self.view._getStageLimitsXY()
```

Comment:
You cannot put this code in this class, as it doesn't have a `.view` attribute. You should put this on miccanvas.DblMicroscopeCanvas. I think `.recenter_buffer()` would be even better anyway, as it'd also catch cases of moving the canvas in different ways, like using the keyboard.